src/net.c

```diff
@@ -205,6 +210,84 @@ int net_get_default_gateway(int family, addr_record_t *record) {
 	closesocket(sock);
 	return -1;
 
+#elif defined(__APPLE__)
+
+	// macOS always uses 4-byte alignment for sockaddrs following message header
+	#define ROUNDUP(a) \
+		((a) > 0 ? (1 + (((a) - 1) | (sizeof(uint32_t) - 1))) : sizeof(uint32_t))
+
+	#define NET_MIB_INTS 6
+	int mib[NET_MIB_INTS] = {CTL_NET, PF_ROUTE, 0, family, NET_RT_FLAGS, RTF_GATEWAY};
+
+	size_t buf_len = 0;
+	if (sysctl(mib, NET_MIB_INTS, 0, &buf_len, 0, 0) != 0) {
+		PLUM_LOG_WARN("sysctl[1] failed");
+		return -1;
+	}
+	if (buf_len == 0) {
+		PLUM_LOG_WARN("sysctl[1] returned 0 buffer length");
+		return -1;
+	}
+	char *buf = (char*)malloc(buf_len);
+	if (sysctl(mib, NET_MIB_INTS, buf, &buf_len, 0, 0) != 0) {
+		PLUM_LOG_WARN("sysctl[2] failed");
+		free(buf);
+		return -1;
+	}
+	
+	int ret = -1;
+	struct rt_msghdr *rtm = NULL;
+	char *buf_limit = (buf + buf_len);
+	for (char *curr = buf; curr < buf_limit; curr += rtm->rtm_msglen) {
+		rtm = (struct rt_msghdr *)curr;
+
+		if ( ((rtm->rtm_flags & (RTF_UP|RTF_GATEWAY)) != (RTF_UP|RTF_GATEWAY))
+			|| ((rtm->rtm_addrs & (RTA_DST|RTA_GATEWAY)) != (RTA_DST|RTA_GATEWAY)) ) {
+			continue;
+		}
+
+		struct sockaddr *sa = (struct sockaddr *)(rtm + 1);
+		struct sockaddr *sa_dst = NULL;
+		struct sockaddr *sa_gateway = NULL;
+		for (int i = 0; i < RTAX_MAX; i++) {
+			if (rtm->rtm_addrs & (1 << i)) {
+				switch (i) {
+				case RTAX_DST:
+					sa_dst = sa;
+					break;
+				case RTAX_GATEWAY:
+					sa_gateway = sa;
+					break;
+				}
+				sa = (struct sockaddr *)((char *)sa + ROUNDUP(sa->sa_len));
+			}
+		}
+
+		if (sa_dst->sa_family != family || sa_gateway->sa_family != family) {
+			continue;
+		}
+
+		if (family == AF_INET) {
+			if (((struct sockaddr_in *)sa_dst)->sin_addr.s_addr == 0) {
```

Comment:
It would be better to compare the `INADDR_ANY` constant instead of the value 0.
```suggestion
			if (((struct sockaddr_in *)sa_dst)->sin_addr.s_addr == INADDR_ANY) {
```